src/service/config/configuration-service.ts

```diff
@@ -0,0 +1,107 @@
+
+    constructor() {
+        switch (Container.get(constants.CONTAINER.ENTRY_POINT)) {
+            case EntryPoint.CLI:
+                this.configuration = new CLIConfiguration();
+                break;
+            case EntryPoint.GITHUB_EVENT:
+                this.configuration = new ActionConfiguration();
+                break;
+            default:
+                logAndThrow("Invalid entrypoint. Please contact with the administrator or report and issue to build-chain tool repository");
+        }
+    }
+
+    /**
+     * Load all necessary data for the configuration object
+     */
+    async init() {
+        await this.configuration.init();
+    }
+
+    /**
+     * Get the name of the project starting the build-chain
+     * @returns {string}
+     */
+    getStarterProjectName(): string {
+        const startProject = this.configuration.parsedInputs.startProject ?? process.env.GITHUB_REPOSITORY;
+        if (startProject) {return startProject;}
+        logAndThrow("Start project needs to be defined or build chain must be run in a Github environment");
+    }
+
+    /**
+     * Check whether the given node is the starter project
+     * @param node 
+     * @returns {Boolean} true if the node is the starter project
+     */
+    isNodeStarter(node: Node): boolean {
+        return node.project === this.getStarterProjectName();
+    }
+
+    /**
+     * Finds the starter node
+     * @returns {Node} starter node
+     */
+    getStarterNode(): Node {
+        const starterNode = this.configuration.projectList.find(node => this.isNodeStarter(node));
+        if (!starterNode) {
+            const errorMessage = `There's no project ${this.getStarterProjectName()} in the chain. This is normally due the project starting 
+                                  the job (or the one selected to behave like so it's not in the project tree information. Please choose a 
+                                  different project like starter or define the project ${this.getStarterProjectName()} in the tree.`;
+            logAndThrow(errorMessage);
+        }
+        return starterNode;
+    }
+
+    /**
+     * Gets the execution level (current, upstream or downstream) for the given node
+     * @param node 
+     * @returns {NodeExecutionLevel} Upstream, current or downstream
+     */
+    getNodeExecutionLevel(node: Node): NodeExecutionLevel {
+        const starterNodeIndex = this.configuration.projectList.indexOf(this.getStarterNode());
+        const currentNodeIndex = this.configuration.projectList.indexOf(node);
+        return currentNodeIndex < starterNodeIndex ? NodeExecutionLevel.UPSTREAM : currentNodeIndex > starterNodeIndex ? NodeExecutionLevel.DOWNSTREAM : NodeExecutionLevel.CURRENT;
+    }
+
+    /**
+     * Checks whether execution needs to be skipped for the given node
+     * @param node 
+     * @returns {Boolean} true if execution needs to be skipped otherwise false
+     */
+    skipExecution(node: Node): boolean {
```

Comment:
This way of inline conditions is old fashion way, now we have newer nodejs and ES versions we can simplify it by 
```suggestion
    skipExecution(node: Node): boolean {
           return this.configuration.parsedInputs.skipExecution ?? this.configuration.parsedInputs.skipProjectExecution ?
                this.configuration.parsedInputs.skipProjectExecution.includes(node.project) :
                false;
```

Reply:
In this case I don't think it will be right to use the null coalescing operator. Null coalescing checks whether `skipExecution` is `null` or `undefined`. However `skipExecution` is never `null` or `undefined`. If `skipExecution` is not defined by the user, it is by default `false` in which case the null coalescing operator will always return `skipExecution` instead of checking for `skipProjectExecution`